pkg/cfn/builder/vpc.go

```diff
@@ -71,9 +76,24 @@ func (c *ClusterResourceSet) addSubnets(refRT *gfn.Value, topology api.SubnetTop
 	}
 }
 
+func (c *ClusterResourceSet) addNATGateways() {
```

Comment:
nitpick: since this is more of a detail than a high level function, can we move it below `addResourcesForVPC()` where it is used?